internal/integrationtest/board/board_test.go

```diff
@@ -0,0 +1,483 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2022 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package board_t_test
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/arduino/arduino-cli/internal/integrationtest"
+	"github.com/stretchr/testify/require"
+	semver "go.bug.st/relaxed-semver"
+	"go.bug.st/testifyjson/requirejson"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+func TestBoardList(t *testing.T) {
+	if os.Getenv("CI") != "" {
+		t.Skip("VMs have no serial ports")
+	}
+
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+	stdout, _, err := cli.Run("board", "list", "--format", "json")
+	require.NoError(t, err)
+	// check is a valid json and contains a list of ports
+	requirejson.Query(t, stdout, ".[] | .port | .protocol!=\"\"", "true")
+	requirejson.Query(t, stdout, ".[] | .port | .protocol_label!=\"\"", "true")
+}
+
+func TestBoardListWithInvalidDiscovery(t *testing.T) {
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+	_, _, err = cli.Run("board", "list")
+	require.NoError(t, err)
+
+	// check that the CLI does not crash if an invalid discovery is installed
+	// (for example if the installation fails midway).
+	// https://github.com/arduino/arduino-cli/issues/1669
+	toolDir := cli.DataDir().Join("packages", "builtin", "tools", "serial-discovery")
+	dirToRemove, err := toolDir.ReadDir()
```

Comment:
@cmaglie originally, this was `dirToRemove, err := toolDir.ReadDirRecursive()` to read the content of each sub-directory too and the `for` cycle included a check to skip the element 0 of the slice. I changed it since the removal of the files kept failing on the `windows-runner` because it didn't have access to the files, somehow. The modified version failed for a few times too, but now it's passing the tests. Let me know if you want me to change it back to how it was.

Reply:
I'm reverting the changes since the problem was solved in PR #1905.